crates/stages/src/stages/hashing_storage.rs

```diff
@@ -62,38 +63,72 @@ impl<DB: Database> Stage<DB> for StorageHashingStage {
             tx.clear::<tables::HashedStorage>()?;
             tx.commit()?;
 
-            let mut first_key = None;
+            let mut current_key = None;
+            let mut current_subkey = None;
+            let mut keccak_address = None;
+
             loop {
-                let next_key = {
+                let mut hashed_batch = BTreeMap::new();
+                let mut remaining = self.commit_threshold as usize;
+                {
                     let mut storage = tx.cursor_dup_read::<tables::PlainStorageState>()?;
+                    while !remaining.is_zero() {
+                        hashed_batch.extend(
+                            storage.walk_dup(current_key, current_subkey)?.take(remaining).map(
+                                |res| {
+                                    let (address, slot) = res.expect("should not fail");
+
+                                    // Address caching for the first iteration when current_key is
+                                    // None
+                                    let keccak_address =
+                                        if let Some(keccak_address) = keccak_address {
+                                            keccak_address
+                                        } else {
+                                            keccak256(address)
+                                        };
+
+                                    // TODO cache map keccak256(slot.key) ?
```

Comment:
agreed

Reply:
but i feel it only makes sense if the cache hashing algo is so much faster than keccak256. 

i'd leave it for the benchmark PR and try it out there

ref #1301